### PR TITLE
FEAT: Facilitate passing tailored solver in params

### PR DESCRIPTION
**Pull request by IvarStefansson**
## Proposed changes

Allows passing a tailored Solver class in params for use in the three functions in run_models.py

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.
- [x] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag. NA

### CODE REVIEW

File: src/porepy/models/run_models.py [hunk 1]
```diff
@@ -249,3 +233,22 @@ def time_step() -> None:
...
+def _choose_solver(model, params: dict) -> Union[pp.LinearSolver, pp.NewtonSolver]:
+    """Choose between linear and non-linear solver.
+
+    Parameters:
+        model: Model class containing all information on parameters, variables,
+            discretization, geometry. Various methods such as those relating to solving
+            the system, see the appropriate solver for documentation.
+        params: Parameters related to the solution procedure.
+
+    """
```
Comment (keileg) [inline src/porepy/models/run_models.py:247]: I think the documentation is fair for the time being. It should be updated as we gradually understand how the interface to the solver should be (as we have managed to do with the methods `prepare_simulation`, `before_nonlinear_iteration` etc. in the models), but at this stage, such abstractions are premature - I just felt like making a note of this somewhere.

**Review by keileg — APPROVED** [review]
Let's include this and refine as we gain more experience.

File: src/porepy/models/run_models.py [hunk 2]
```diff
@@ -249,3 +233,22 @@ def time_step() -> None:
                 time_progressbar.update(n=model._time_step / initial_time_step)
 
     model.after_simulation()
+
+
+def _choose_solver(model, params: dict) -> Union[pp.LinearSolver, pp.NewtonSolver]:
+    """Choose between linear and non-linear solver.
+
+    Parameters:
+        model: Model class containing all information on parameters, variables,
```
Comment (IvarStefansson) [inline src/porepy/models/run_models.py:242]:
```suggestion
        model: Model class containing all information on material parameters, variables,
```

File: src/porepy/models/run_models.py [hunk 3]
```diff
@@ -249,3 +233,22 @@ def time_step() -> None:
...
+            discretization, geometry. Various methods such as those relating to solving
```
Comment (IvarStefansson) [inline src/porepy/models/run_models.py:243]:
```suggestion
            discretization and geometry. Various methods such as those relating to solving
```

File: src/porepy/models/run_models.py (hunk 1 above)
Comment (IvarStefansson) [reply src/porepy/models/run_models.py:247]: Sure! I have two minor suggestions, please commit and merge if you approve.

**Pull request merged**